gnosis/verification.k

```diff
@@ -99,31 +116,69 @@ module VERIFICATION
     rule 0 <=Int #bufElm(_, _)  => true
     rule #bufElm(_, _) <Int 256 => true
 
+    rule 0 <=Int X xorInt 115792089237316195423570985008687907853269984665640564039457584007913129639935 => true      requires 0<=Int X andBool X <Int pow256
+
+    rule X xorInt 115792089237316195423570985008687907853269984665640564039457584007913129639935 <Int pow256 => true  requires 0<=Int X andBool X <Int pow256 
+
+
   // ########################
   // Simplification
   // ########################
 
+    rule (X +Int 31) /Int 32 *Int 32 => #ceil32(X)
+    rule (31 +Int X) /Int 32 *Int 32 => #ceil32(X)
+
+    rule #asWord(#buf(32, DATA)) => DATA  requires #rangeUInt(256, DATA)
+
     rule #padToWidth(32, #asByteStack(V)) => #buf(32, V) // #asByteStackInWidth(V, 32)
-      requires 0 <=Int V andBool V <Int pow256
+      requires 0 <=Int V andBool V <Int pow256 andBool #getKLabelString(V) =/=String "#asWord"
 
+    rule #buf(32, #asWord(WS)) => WS  requires #sizeWordStack(WS) ==Int 32
     rule #buf(32, #asWord(#bufSeg(BUF, START, WIDTH))) => #bufSeg(BUF, START +Int WIDTH -Int 32, 32)  requires WIDTH >=Int 32
 
     rule #asWord(#bufSeg(BUF, START, WIDTH)) &Int 255 => #asWord(#bufSeg(BUF, START +Int WIDTH -Int 1, 1))  requires WIDTH >=Int 1
     rule 255 &Int #asWord(#bufSeg(BUF, START, WIDTH)) => #asWord(#bufSeg(BUF, START +Int WIDTH -Int 1, 1))  requires WIDTH >=Int 1
 
     rule #bufSeg(BUF, START, 1) => #bufElm(BUF, START) : .WordStack
+    rule #asWord(#bufElm(BUF, INDEX) : .WordStack) => #bufElm(BUF, INDEX)
+
+    rule #buf(32, DATA) => #padToWidth(32, #asByteStack(DATA))  [concrete]
 
     rule #asWord(WS0 ++ #buf(MASKWIDTH0, 0)) |Int #asWord(#buf(MASKWIDHT1, 0) ++ WS1)
       => #asWord(WS0 ++ WS1)
       requires #sizeWordStack(WS0) +Int MASKWIDTH0 ==Int 32
        andBool #sizeWordStack(WS0) ==Int MASKWIDHT1
        andBool MASKWIDTH0 ==Int #sizeWordStack(WS1)
 
-    // Arithmetic
+    rule #asWord(WS) &Int 115792089237316195423570985008687907853269984665640564039457584007913129639935 => #asWord(WS)
```

Comment:
again, please comment for the big number

Reply:
done